### ISSUE TITLE
Build font with API example

### ISSUE THREAD

**Issue by tijmenvangurp** [fabiospampinato/icon-font-buildr#4]
Hey, first of all, nice work on the script. This is the only one that I found that supports ligatures nicely. 

The command line works, but on you API example from the readme file I get a few warnings that I cannot easily resolve myself.  

`SyntaxError: await is only valid in async function`

Would it be possible for you to add an API example to the repo?

**Comment by fabiospampinato** [comment]
I'll try to improve upon our [current example](https://github.com/fabiospampinato/icon-font-buildr#api). Unfortunately I'm not using the API in any project that I can link to.

Your specific error is about the `await` keyword, which is used at the "top level", and not inside an `async` function. This isn't supported by your Node.js version (I'm not sure if any version of Node supports yet, but there's a [stage-2 proposal](https://github.com/tc39/proposal-top-level-await) about this) so you're getting an error.

**Comment by tijmenvangurp** [comment]
Changed it to 

```
async function asyncCall() {
    console.log('calling');
    var result = await  builder.build()
    console.log('done')
    const codepoints = builder.getIconsCodepoints (); // Get a map of icon names to codepoints, useful for generating HTML/CSS/SCSS etc.

    console.log(codepoints);
  }
  
  asyncCall();
```
Getting error now  `(node:48079) UnhandledPromiseRejectionWarning: TypeError: this.config.icons.reduce is not a function`

using node v10.15.3

**Comment by tijmenvangurp** [comment]
One of the reasons why I want to use the API is to create css file for development with css classes and their codepoints. Was hoping I could use the getIconsCodePoints for this.

**Comment by fabiospampinato** [comment]
I'll make sure the example in the readme actually works. In the meantime you can check the code used for testing here: https://github.com/fabiospampinato/icon-font-buildr/blob/master/test/index.js

**Comment by tijmenvangurp** [comment]
`console.log(builder.config.icons);`

Returns 
```
{ backup:
   { icon: 'backup',
     name: 'backup',
     codepoints: [ '' ],
     ligatures: [ 'backup' ] },
  bug_report:
   { icon: 'bug_report',
     name: 'bug_report',
     codepoints: [ '' ],
     ligatures: [ 'bug_report' ] },
  amazon:
   { icon: 'amazon',
     name: 'amazon',
     codepoints: [ '' ],
     ligatures: [ 'amazon' ] },
  'android-debug-bridge':
   { icon: 'android-debug-bridge',
     name: 'android debug icon',
     codepoints: [ '', '' ],
     ligatures: [ 'debug', 'debug_alt' ] } }
```

how is `builder.getIconsCodepoints();` different from this object?

**Comment by tijmenvangurp** [comment]
Not pritty but this works to create codepoint overview. Only need to write it to a scss file.

```
async function asyncCall() {
    console.log('starting to compile');
    var result = await builder.build();
    console.log('done!');
    // const codepoints = builder.getIconsCodepoints(); // Get a map of icon names to codepoints, useful for generating HTML/CSS/SCSS etc.
    console.log(builder.config.output.icons);

    for (const iconName in builder.config.icons) {
        if (builder.config.icons.hasOwnProperty(iconName)) {
            const iconObject = builder.config.icons[iconName];
            console.log(iconName);
            for (let i = 0; i < iconObject.codepoints.length; i++) {
                const element = iconObject.codepoints[i];
                console.log(element.charCodeAt(0).toString(16));
                
            }
        }
    }
}

asyncCall();

```

output:
```
backup
e000
bug_report
e001
amazon
e002
android-debug-bridge
e042
e064
```

**Comment by fabiospampinato** [comment]
I've fixed the issues in v1.3.3.

> how is builder.getIconsCodepoints(); different from this object?

It returns this:

```js
{ 
  backup: [ '' ],
  bug_report: [ '' ],
  amazon: [ '' ],
  'android-debug-bridge': [ '', '' ] 
}
```

Or hex values for those codepoints if it's called like this: `getIconsCodepoints ( true );`. It provides a stable API that doesn't rely on the internals of the library.
